code/modules/mob/typing_indicator.dm

```diff
@@ -1,31 +1,30 @@
-/mob/proc/toggle_typing_indicator(emoting)
-	var/image/typing_indicator
+/mob/proc/add_typing_indicator(emoting)
+	if(stat != CONSCIOUS)
+		return
 
 	if(emoting)
 		typing_indicator = image('icons/mob/talk.dmi', src, "emoting")
 	else
 		typing_indicator = image('icons/mob/talk.dmi', src, "typing")
 
-	if(!(client?.prefs?.show_typing))
-		remove_emote_overlay(client, typing_indicator, viewers())
-		return
+	add_emote_overlay(typing_indicator, remove_delay = NONE)
 
-	if(typing)
-		remove_emote_overlay(client, typing, viewers())
-		typing = null
 
-	else if(stat == CONSCIOUS)
-		typing = typing_indicator
-		add_emote_overlay(typing, remove_delay = NONE)
+/mob/proc/remove_typing_indicator(emoting)
+	if(!typing_indicator)
+		return
+
+	remove_emote_overlay(client, typing_indicator, viewers())
```

Comment:
What about the people that were around when it triggered but moved away for the removal?
Do you need to keep a list of "viewers" to ensure you remove the emote from the right people instead?

Reply:
So far it hasn't been an issue, I am not changing the behavior.

Reply:
There are definite issues with the indicator not going away, this seems like its an area for improvement.

Reply:
I am fixing one specific issue, feel free to fix more!